shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -204,16 +204,17 @@ - (void)setSystemChromeSystemUIOverlayStyle:(NSDictionary*)message {
 
 - (void)popSystemNavigator:(BOOL)isAnimated {
   // Apple's human user guidelines say not to terminate iOS applications. However, if the
-  // root view of the app is a navigation controller, it is instructed to back up a level
-  // in the navigation hierarchy.
+  // FlutterViewController or one of its ancestors is a child of a navigation controller,
+  // it is instructed to back up a level in the navigation hierarchy.
   // It's also possible in an Add2App scenario that the FlutterViewController was presented
   // outside the context of a UINavigationController, and still wants to be popped.
-  UIViewController* viewController = [UIApplication sharedApplication].keyWindow.rootViewController;
-  if ([viewController isKindOfClass:[UINavigationController class]]) {
-    [((UINavigationController*)viewController) popViewControllerAnimated:isAnimated];
+  auto engineViewController = static_cast<UIViewController*>([_engine.get() viewController]);
+  if ([engineViewController navigationController] != nil) {
```

Comment:
Nit: Maybe `if (auto navitationController = engineViewController.navigationController)` instead in case the navigation controller property is computed.

Reply:
Looks good! Thanks